spec/unit/module_spec.rb

```diff
@@ -690,6 +690,30 @@ def a_module_with_metadata(data)
     end
   end
 
+  describe "when loading the metadata file from disk" do
+    it "should properly parse utf-8 contents" do
```

Comment:
@joshcooper I created some real files on disk in `UTF-16LE` and `UTF-16LE` with BOM (which should approximate `UCS-2`) and added a couple more similar tests locally. Since Ruby does no detection when opening files to read them, the results were what I expected (against both the previous and updated code) - unparseable JSON.  So no change in behavior there:

```
1) Puppet::Module when loading the metadata file from disk should properly parse utf-16le contents
     Failure/Error: @metadata = data = JSON.parse(File.read(metadata_file))

     JSON::ParserError:
       757: unexpected token at '{'
     # ./.bundle/gems/ruby/2.1.0/gems/json-1.8.3/lib/json/common.rb:155:in `parse'
     # ./.bundle/gems/ruby/2.1.0/gems/json-1.8.3/lib/json/common.rb:155:in `parse'
     # ./lib/puppet/module.rb:151:in `load_metadata'
     # ./spec/unit/module_spec.rb:698:in `block (3 levels) in <top (required)>'

  2) Puppet::Module when loading the metadata file from disk should properly parse utf-16le with BOM contents
     Failure/Error: @metadata = data = JSON.parse(File.read(metadata_file))

     JSON::ParserError:
       757: unexpected token at '??{'
     # ./.bundle/gems/ruby/2.1.0/gems/json-1.8.3/lib/json/common.rb:155:in `parse'
     # ./.bundle/gems/ruby/2.1.0/gems/json-1.8.3/lib/json/common.rb:155:in `parse'
     # ./lib/puppet/module.rb:151:in `load_metadata'
     # ./spec/unit/module_spec.rb:706:in `block (3 levels) in <top (required)>'
```

I will try to construct some other sample files to verify old / new behavior.